app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/MainActivity.kt

```diff
@@ -0,0 +1,493 @@
+                return true
+            }
+            R.id.menu_item_favorites -> {
+                openFavorites()
+                return true
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun openSessionDetails(sessionId: String) {
+        if (appRepository.updateSelectedSessionId(sessionId)) {
+            val sidePaneView = findViewById<FragmentContainerView>(R.id.detail)
+            if (sidePaneView == null) {
+                SessionDetailsActivity.startForResult(this)
+            } else {
+                SessionDetailsFragment.replaceAtBackStack(supportFragmentManager, R.id.detail, true)
+            }
+        }
+    }
+
+    override fun onSidePaneClose(fragmentTag: String) {
+        findViewById<View>(R.id.detail)?.let {
+            it.isVisible = false
+        }
+        if (fragmentTag == StarredListFragment.FRAGMENT_TAG) {
+            isFavoritesInSidePane = false
+        }
+        removeFragment(fragmentTag)
+    }
+
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
+        super.onActivityResult(requestCode, resultCode, intent)
+        when (requestCode) {
+
+            AlarmList.REQUEST_CODE,
+            SessionDetailsActivity.REQUEST_CODE ->
+                if (resultCode == RESULT_CANCELED) {
+                    shouldScrollToCurrent = false
+                }
+
+            SettingsActivity.REQUEST_CODE ->
+                if (resultCode == RESULT_OK && intent != null) {
+                    val isAlternativeHighlightingUpdated = intent.getBooleanExtra(
+                        BUNDLE_KEY_ALTERNATIVE_HIGHLIGHTING_UPDATED, false)
+                    val isUseDeviceTimeZoneUpdated = intent.getBooleanExtra(
+                        BUNDLE_KEY_USE_DEVICE_TIME_ZONE_UPDATED, false)
+
+                    if (isAlternativeHighlightingUpdated || isUseDeviceTimeZoneUpdated) {
+                        if (findViewById<View>(R.id.schedule) != null && findFragment(FahrplanFragment.FRAGMENT_TAG) == null) {
+                            replaceFragment(R.id.schedule, FahrplanFragment(), FahrplanFragment.FRAGMENT_TAG)
+                        }
+                    }
+                    var shouldFetchFahrplan = false
+                    val isScheduleUrlUpdated = resources.getBoolean(
+                        R.bool.bundle_key_schedule_url_updated_default_value)
+                    if (intent.getBooleanExtra(BUNDLE_KEY_SCHEDULE_URL_UPDATED, isScheduleUrlUpdated)) {
+                        shouldFetchFahrplan = true
+                    }
+                    val isEngelsystemShiftsUrlUpdated = resources.getBoolean(
+                        R.bool.bundle_key_engelsystem_shifts_url_updated_default_value)
+                    if (intent.getBooleanExtra(BUNDLE_KEY_ENGELSYSTEM_SHIFTS_URL_UPDATED, isEngelsystemShiftsUrlUpdated)) {
+                        shouldFetchFahrplan = true
+                    }
```

Comment:
The if conditions can be merged:

```
val updateIntent = intent.getBooleanExtra(BUNDLE_KEY_SCHEDULE_URL_UPDATED, isScheduleUrlUpdated)
val isEngelsystemShiftsUrlUpdated = resources.getBoolean(      R.bool.bundle_key_engelsystem_shifts_url_updated_default_value)
val updateEngelSystem = intent.getBooleanExtra(BUNDLE_KEY_SCHEDULE_URL_UPDATED, isScheduleUrlUpdated)

val shouldFetchFahrplan = updateIntent || updateEngelSystem
```

Reply:
:x: I would not merge both blocks because they associate with separate/independent options in the Settings screen. That would mix two contextual separate features into one code block. (BTW: You mixed up some variables in your snippet).